src/main/datatypes/MQ2GroupType.cpp

```diff
@@ -414,6 +454,261 @@ bool MQ2GroupType::GetMember(MQVarPtr VarPtr, const char* Member, char* Index, M
+		for(auto& member : *pLocalPC->Group)
+		{
+			if(member
+			   && member->Type == EQP_PC
+			   && member->pSpawn
+			   && member->pSpawn->GetClass() == Monk)
+			{
+				Dest = pSpawnType->MakeTypeVar(member->pSpawn);
+				return true;
+			}
+		}
+
+		Dest = pSpawnType->MakeTypeVar();
+		return true;
+
+	case GroupMembers::Rogue:
+		for(auto& member : *pLocalPC->Group)
+		{
+			if(member
+			   && member->Type == EQP_PC
+			   && member->pSpawn
+			   && member->pSpawn->GetClass() == Rogue)
+			{
+				Dest = pSpawnType->MakeTypeVar(member->pSpawn);
+				return true;
+			}
+		}
+
+		Dest = pSpawnType->MakeTypeVar();
+		return true;
+
+	case GroupMembers::Necromancer:
+		for(auto& member : *pLocalPC->Group)
+		{
+			if(member
+			   && member->Type == EQP_PC
+			   && member->pSpawn
+			   && member->pSpawn->GetClass() == Necromancer)
+			{
+				Dest = pSpawnType->MakeTypeVar(member->pSpawn);
+				return true;
+			}
+		}
+
+		Dest = pSpawnType->MakeTypeVar();
+		return true;
+
+	case GroupMembers::Wizard:
+		for(auto& member : *pLocalPC->Group)
+		{
+			if(member
+			   && member->Type == EQP_PC
+			   && member->pSpawn
+			   && member->pSpawn->GetClass() == Wizard)
+			{
+				Dest = pSpawnType->MakeTypeVar(member->pSpawn);
+				return true;
+			}
+		}
+
+		Dest = pSpawnType->MakeTypeVar();
+		return true;
+
+	case GroupMembers::Mage:
+		for(auto& member : *pLocalPC->Group)
+		{
+			if(member
+			   && member->Type == EQP_PC
+			   && member->pSpawn
+			   && member->pSpawn->GetClass() == Mage)
+			{
+				Dest = pSpawnType->MakeTypeVar(member->pSpawn);
+				return true;
+			}
+		}
+
+		Dest = pSpawnType->MakeTypeVar();
+		return true;
+
+	case GroupMembers::Magician:
+		for(auto& member : *pLocalPC->Group)
+		{
+			if(member
+			   && member->Type == EQP_PC
+			   && member->pSpawn
+			   && member->pSpawn->GetClass() == Magician)
+			{
+				Dest = pSpawnType->MakeTypeVar(member->pSpawn);
+				return true;
+			}
+		}
+
+		Dest = pSpawnType->MakeTypeVar();
+		return true;
+
+	case GroupMembers::Enchanter:
+		for(auto& member : *pLocalPC->Group)
+		{
+			if(member
+			   && member->Type == EQP_PC
+			   && member->pSpawn
+			   && member->pSpawn->GetClass() == Enchanter)
+			{
+				Dest = pSpawnType->MakeTypeVar(member->pSpawn);
+				return true;
+			}
+		}
+
+		Dest = pSpawnType->MakeTypeVar();
+		return true;
+
+	case GroupMembers::Beastlord:
+		for(auto& member : *pLocalPC->Group)
+		{
+			if(member
+			   && member->Type == EQP_PC
+			   && member->pSpawn
+			   && member->pSpawn->GetClass() == Beastlord)
+			{
+				Dest = pSpawnType->MakeTypeVar(member->pSpawn);
+				return true;
+			}
+		}
+
+		Dest = pSpawnType->MakeTypeVar();
+		return true;
+
+	case GroupMembers::Berserker:
+		for(auto& member : *pLocalPC->Group)
+		{
+			if(member
+			   && member->Type == EQP_PC
+			   && member->pSpawn
+			   && member->pSpawn->GetClass() == Berserker)
+			{
+				Dest = pSpawnType->MakeTypeVar(member->pSpawn);
+				return true;
+			}
+		}
+
```

Comment:
If you ever find yourself copy pasting the same code 13 times, that should be a huge red flag.